### PR TITLE
Ginkgo: Test Flake on RuntimeValidatedConntrack

### PR DESCRIPTION
**Pull request by eloycoto**
This test was failed randonmly in the last two days because pods were
not ready after a new change was applied.

Test log:

```
level=debug msg="running command: sudo cilium config PolicyEnforcement=default"
cmd: "sudo cilium config PolicyEnforcement=default" exitCode: 0

level=debug msg="running command: sudo cilium endpoint list -o jsonpath='{range [?(@.labels.orchestration-identity[0]!=\"reserved:health\")]}{@.container-name}{\"=\"}{@.id}{\"\\n\"}{end}'"
cmd: "sudo cilium endpoint list -o jsonpath='{range [?(@.labels.orchestration-identity[0]!=\"reserved:health\")]}{@.container-name}{\"=\"}{@.id}{\"\\n\"}{end}'" exitCode: 0
 client=12075
server=14594
server-2=19080
server-3=20084
netcat=27624
client-2=28707

level=debug msg="running command: sudo cilium endpoint config 28707 IngressPolicy=false"
cmd: "sudo cilium endpoint config 28707 IngressPolicy=false" exitCode: 0
 Endpoint 28707 configuration updated successfully

level=debug msg="running command: sudo cilium endpoint config 28707 EgressPolicy=false"
cmd: "sudo cilium endpoint config 28707 EgressPolicy=false" exitCode: 0
 Endpoint 28707 configuration updated successfully

level=debug msg="running command: sudo cilium endpoint config 12075 IngressPolicy=false"
cmd: "sudo cilium endpoint config 12075 IngressPolicy=false" exitCode: 0
 Endpoint 12075 configuration updated successfully

level=debug msg="running command: sudo cilium endpoint config 12075 EgressPolicy=false"
cmd: "sudo cilium endpoint config 12075 EgressPolicy=false" exitCode: 0
 Endpoint 12075 configuration updated successfully

level=debug msg="running command: sudo cilium endpoint config 14594 IngressPolicy=false"
cmd: "sudo cilium endpoint config 14594 IngressPolicy=false" exitCode: 1
 Error: Cannot update endpoint 14594: [PATCH /endpoint/{id}/config][500] patchEndpointIdConfigFailed  unable to regenerate endpoint program because state transition to waiting-to-regenerate was unsuccessful; check `cilium endpoint log 14594` for more information
```

This error happened on:

- https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/1633/testReport/
- https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/1632/testReport/
- https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/1631/testReport/
- https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/1630/testReport/

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

File: test/runtime/ct.go [hunk 1]
```diff
@@ -619,15 +619,20 @@ var _ = Describe("RuntimeValidatedConntrackTable", func() {
...
+		Expect(areEndpointsReady).Should(BeTrue(), "Endpoints are not ready after timeout")
+
 		// #FIXME remove these 6 lines once GH-2496 is fixed
 		epIDs, err := vm.GetEndpointsIds()
 		Expect(err).To(BeNil(), "Getting endpoints identity IDs")
 		for _, v := range epIDs {
-			vm.ExecCilium(fmt.Sprintf("endpoint config %s IngressPolicy=false", v)).ExpectSuccess("Setting %s endpoint's IngressPolicy as false", v)
-			vm.ExecCilium(fmt.Sprintf("endpoint config %s EgressPolicy=false", v)).ExpectSuccess("Setting %s endpoint's EgressPolicy as false", v)
+			vm.ExecCilium(fmt.Sprintf("endpoint config %s IngressPolicy=false", v)).ExpectSuccess(
+				"Setting %s endpoint's IngressPolicy as false", v)
```
Comment (ianvernon) [inline test/runtime/ct.go]: https://github.com/cilium/cilium/pull/3054 was merged to unblock the CI. What I think should be done to fix this issue once the correct functionality that was actually reverted in https://github.com/cilium/cilium/pull/3054 is restored, is after each configuration update for an endpoint, wait until the endpoint is regenerated. The only wrinkle there though, is sometimes the endpoint does not actually have to be regenerated :) I think we can get around this by checking the output of `cilium endpoint config` for each endpoint and then if the option does need to be changed, change it and then wait for regeneration.

Comment (ianvernon) [reply test/runtime/ct.go]: After each configuration update, add: 
```
areEndpointsReady = vm.WaitEndpointsReady()
Expect(areEndpointsReady).Should(BeTrue(), "Endpoints not ready after timeout")
```

I think we should eventually factor config updates for endpoints / cilium into helper functions which wait for endpoints to be in ready state after applying such updates.

**Review by ianvernon — CHANGES_REQUESTED** [review]

**Comment by eloycoto** [comment]
test-me-please

File: test/runtime/ct.go [hunk 2]
```diff
@@ -619,15 +622,21 @@ var _ = Describe("RuntimeValidatedConntrackTable", func() {
...
+		areEndpointsReady := vm.WaitEndpointsReady()
+		Expect(areEndpointsReady).Should(BeTrue(), "Endpoints are not ready after timeout")
+
 		// #FIXME remove these 6 lines once GH-2496 is fixed
 		epIDs, err := vm.GetEndpointsIds()
 		Expect(err).To(BeNil(), "Getting endpoints identity IDs")
 		for _, v := range epIDs {
-			vm.ExecCilium(fmt.Sprintf("endpoint config %s IngressPolicy=false", v)).ExpectSuccess("Setting %s endpoint's IngressPolicy as false", v)
-			vm.ExecCilium(fmt.Sprintf("endpoint config %s EgressPolicy=false", v)).ExpectSuccess("Setting %s endpoint's EgressPolicy as false", v)
+			status := vm.EndpointSetConfig(v, helpers.OptionIngressPolicy, "false")
```
Comment (ianvernon) [inline test/runtime/ct.go:632]: I forgot we had this function ;) great! Thanks for taking care of this.

**Review by ianvernon — APPROVED** [review]

**Pull request merged**